pandas/tests/scalar/test_period.py

```diff
@@ -1406,3 +1406,14 @@ def test_period_ops_offset(self):
 
         with tm.assert_raises_regex(period.IncompatibleFrequency, msg):
             p - offsets.Hour(2)
+
+
+def test_period_immutable():
+    # see gh-17116
+    per = pd.Period('2014Q1')
+    with pytest.raises(AttributeError, message="is not writable"):
```

Comment:
the `message="is not writable"` is not needed here, the default assertion message of pytest is normally clear.

If you actually want to check the message of the AttributeError, it is the `match` keyword you need to use, not `message` (but I don't think it is needed to test that, it is not a custom error message we write ourselves)